### DISCUSSION TITLE
Regarding Endless Rotation

### DISCUSSION THREAD

**Discussion by ashishsiddharth** [Yaskawa-Global/YMConnect#17] [General]
I want to perform endless rotation (rpm), along with motions about the tool frame (x,y,z). How do you perform that?
Controller: YRC1000
Robot: GP12

**Comment by ted-miller** [comment]
Hello @ashishsiddharth 
Unfortunately, the endless-rotation feature is not supported by YMConnect and is not planned to be implemented.

Motion in tool frame does not appear to be working currently. We will investigate this and get it resolved.

**Reply by ted-miller** [reply]
After investigating, we have determined that motion in Tool Frame will not be supported. But we will improve this error handling in the next release.

**Comment by ashishsiddharth** [comment]
Hello @ted-miller ,
I tried using High-speed Ethernet Protocol, but that also doesn't seem to work. So I guess I will have to use Motoman SDK or MotoPlus IDE ? If so, is there any forum, documentation, or proper reference to move ahead??

I went through manual number HW1483602, moto plus reference manual. I could not find a reference for endless functions. Please help

**Reply by ted-miller** [reply]
The MotoPlus SDK is the American version of MotoPlus IDE. Functionally, they are the same. 

Unfortunately, none of the motion API's available in MotoPlus (or other SDK) support endless rotation. That feature is reserved exclusively for INFORM programming on the teach pendant. 

I'm not aware of any current plans to support this functionality in the near future.

If you have any other questions about using the MotoPlus SDK, please feel free to email me directly, using the address on my [profile page](https://github.com/ted-miller).
